src/Android/Xamarin.Android/Samples/Map/OpenMobileMap/OpenMobileMap.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2017 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+// language governing permissions and limitations under the License.
+
+using System.Linq;
+using System.IO;
+using Android.App;
+using Android.OS;
+using Android.Widget;
+using Esri.ArcGISRuntime.Mapping;
+using Esri.ArcGISRuntime.UI.Controls;
+using ArcGISRuntimeXamarin.Managers;
+
+
+namespace ArcGISRuntimeXamarin.Samples.OpenMobileMap
+{
+    [Activity]
+    public class OpenMobileMap : Activity
+    {
+        private MapView _myMapView;
+
+        protected override void OnCreate(Bundle bundle)
+        {
+            base.OnCreate(bundle);
+
+            Title = "Open mobile map (map package)";
+
+            // Create the UI, setup the control references and execute initialization
+            CreateLayout();
+            Initialize();
+        }
+
+        private async void Initialize()
+        {
+            // The mobile map package will be downloaded from ArcGIS Online
+            // The data manager (a component of the sample viewer, *NOT* the runtime
+            //     handles the offline data process
+
+            // The desired MMPK is expected to be called Yellowstone.mmpk
+            string filename = "Yellowstone.mmpk";
+
+            // The data manager provides a method to get the folder
+            string folder = DataManager.GetDataFolder();
+
+            // Get the full path
+            string filepath = Path.Combine(folder, "SampleData", "OpenMobileMap", filename);
+
+            // Variable to hold the MobileMapPackage
+            MobileMapPackage myMapPackage;
+
+            // Open the package, try downloading if that fails
```

Comment:
I might avoid using try/catch as logical operators - exceptions are expensive. I'd probably check if the file existed first. And run the appropriate code based of the bool returned. https://msdn.microsoft.com/en-us/library/system.io.file.exists.aspx